### ISSUE TITLE
I think Autokeras doesn't export best model 

### ISSUE THREAD

**Issue by JiHyeoNEE** [keras-team/autokeras#1153]
Hi I'm using autokeras for my study.
I got an exported model from autokeras but I think it is not a best model.
This is my code.

model = ak.AutoModel(inputs=ak.ImageInput(), outputs=ak.ClassificationHead(), tuner='random', max_trials=50)
model.fit(train_input_data, train_target_data, epochs=100, validation_split=0.2, batch_size=1, shuffle=True, verbose=2)
final_model = model.export_model()

And then I evaluated final_model, but it's accuracy was just 0.51 and it's structure was too simple although a lot of complicated models were built during training.

Do you have an idea about that?

autoekeras version : 1.0.2
tensorflow version : tensorflow-gpu 2.1.0

**Comment by achaar** [comment]
Hi, please check Customized Space Search in https://autokeras.com/tutorial/image_classification/ to see if you are defining the model correctly. 
Thanks.

**Comment by JiHyeoNEE** [comment]
@achaar I tried it but result was same.. Thank you for the reply.

**Comment by haifeng-jin** [comment]
I think it might because the metrics are evaluated on different data. During testing, I assume you passed testing data. During fit, AutoKeras split your training data and the evaluation is done on the validation data splited from the training data.
